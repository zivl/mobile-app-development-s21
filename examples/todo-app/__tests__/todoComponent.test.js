import React from 'react';
import { Provider } from 'react-redux';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import Todo from '../src/components/Todo/Todo';
import { store } from '../src/store';
import nock from 'nock';

describe('Todo Component Tests', () => {
  test('should render "no items found" message when there are no todos in list', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Todo />
      </Provider>,
    );
    expect(getByText('No Todo Items Found')).toBeDefined();
  });

  test('should render todo list when items are in list', () => {
    const { getByText, getByTestId } = render(
      <Provider store={store}>
        <Todo />
      </Provider>,
    );
    const inputText = getByTestId('todo-input-text');
    const addButton = getByText('Add');
    const todoText = 'fix car';
    fireEvent.changeText(inputText, todoText);
    fireEvent.press(addButton);
    expect(getByText(todoText)).toBeDefined();
  });

  test('should the same screen snapshot when adding new todo in list', () => {
    const { getByText, getByTestId, toJSON } = render(
      <Provider store={store}>
        <Todo />
      </Provider>,
    );
    const inputText = getByTestId('todo-input-text');
    const addButton = getByText('Add');
    const todoText = 'fix car';
    fireEvent.changeText(inputText, todoText);
    fireEvent.press(addButton);
    expect(toJSON()).toMatchSnapshot();
  });

  test('should return user detail and present on title', async () => {
    const expectedUsername = 'ziv';
    //https://api.soundcloud.com/me?client_id=CW62xLA9h8wXrXC1WIaSX9OWA6novVIE
    // baseUrl / path / query params
    nock('https://api.soundcloud.com')
      .get(/me(.*)/)
      .reply(200, {
        user: {
          name: expectedUsername,
        },
      });
    const { getByText, getByTestId, toJSON } = render(
      <Provider store={store}>
        <Todo />
      </Provider>,
    );

    const loadButton = getByText('Load Data');
    fireEvent.press(loadButton);
    await waitFor(() => expect(getByTestId('username-text')).toBeDefined());
    const textComponent = getByTestId('username-text');
    expect(textComponent.children[0]).toEqual(expectedUsername);
    expect(toJSON()).toMatchSnapshot();
  });
});
