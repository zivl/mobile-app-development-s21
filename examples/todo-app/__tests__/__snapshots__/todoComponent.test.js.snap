// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component Tests should return user detail and present on title 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      testID="username-text"
    >
      ziv
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      clearButtonMode="while-editing"
      onChangeText={[Function]}
      placeholder="what up?"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "fontSize": 20,
          "padding": 10,
        }
      }
      testID="todo-input-text"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          All
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Completed
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Uncompleted
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={
      <Text>
        No Todo Items Found
      </Text>
    }
    data={
      Array [
        Object {
          "id": 0,
          "isDone": false,
          "text": "fix car",
        },
        Object {
          "id": 1,
          "isDone": false,
          "text": "fix car",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "gray",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textDecorationLine": "none",
                    },
                  ]
                }
              >
                fix car
              </Text>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "gray",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textDecorationLine": "none",
                    },
                  ]
                }
              >
                fix car
              </Text>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Todo Component Tests should the same screen snapshot when adding new todo in list 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Load Data
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      clearButtonMode="while-editing"
      onChangeText={[Function]}
      placeholder="what up?"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "fontSize": 20,
          "padding": 10,
        }
      }
      testID="todo-input-text"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          All
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Completed
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Uncompleted
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={
      <Text>
        No Todo Items Found
      </Text>
    }
    data={
      Array [
        Object {
          "id": 0,
          "isDone": false,
          "text": "fix car",
        },
        Object {
          "id": 1,
          "isDone": false,
          "text": "fix car",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "gray",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textDecorationLine": "none",
                    },
                  ]
                }
              >
                fix car
              </Text>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "gray",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    Object {
                      "textDecorationLine": "none",
                    },
                  ]
                }
              >
                fix car
              </Text>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
